symphony-sdk/src/main/java/authentication/AbstractSymExtensionAppAuth.java

```diff
@@ -0,0 +1,112 @@
+    }
+
+    public Boolean validateTokens(String appToken, String symphonyToken) {
+        return tokensRepository.get(appToken)
+                .filter(token -> token.getSymphonyToken().equals(symphonyToken))
+                .isPresent();
+    }
+
+    protected String generateToken() {
+        byte[] randBytes = new byte[64];
+        secureRandom.nextBytes(randBytes);
+        return Hex.encodeHexString(randBytes);
+    }
+
+    protected AppAuthResponse handleSessionAppAuthFailure(Response response, String appToken, String... podSessionAuthUrl) {
+        try {
+            handleError(response, null);
+        } catch (Exception e) {
+            logger.error("Unexpected error, retry authentication in {} seconds", AuthEndpointConstants.TIMEOUT);
+        }
+        try {
+            TimeUnit.SECONDS.sleep(AuthEndpointConstants.TIMEOUT);
+        } catch (InterruptedException e) {
+            logger.error("Error with authentication", e);
+        }
+        if (authRetries++ > AuthEndpointConstants.MAX_AUTH_RETRY) {
+            logger.error("Max retries reached. Giving up on auth.");
+            return null;
+        }
+        return sessionAppAuthenticate(appToken, podSessionAuthUrl);
+    }
+
+    protected String formattedPodSessionAuthUrl(String... podSessionAuthUrl) {
+        String formattedUrl;
+        if (podSessionAuthUrl.length == 0 || podSessionAuthUrl[0] == null
+                || StringUtils.isBlank(podSessionAuthUrl[0])) {
+            if (config == null) {
+                throw new NoConfigException("Must provide a SymConfig object to authenticate");
+            }
+            formattedUrl = this.config.getSessionAuthUrl();
+        } else {
+            formattedUrl = CommonConstants.HTTPS_PREFIX + podSessionAuthUrl[0];
+        }
+        return formattedUrl;
+    }
+
+    protected String getPodCertificateFromCertPath(String podCertPath, String... podSessionAuthUrl) {
+        String target = this.formattedPodSessionAuthUrl(podSessionAuthUrl);
+        String podCertificate;
+        do {
+            Response response
+                    = sessionAuthClient.target(target)
+                    .path(podCertPath)
+                    .request(MediaType.APPLICATION_JSON)
+                    .get();
+            if (response.getStatusInfo().getFamily() != Response.Status.Family.SUCCESSFUL) {
+                try {
+                    handleError(response, null);
+                } catch (Exception e) {
+                    logger.error("Unexpected error, retry authentication in 30 seconds");
+                }
+                try {
+                    TimeUnit.SECONDS.sleep(AuthEndpointConstants.TIMEOUT);
+                } catch (InterruptedException e) {
+                    logger.error("Error with verify", e);
+                }
+            } else {
+                podCertificate = response.readEntity(PodCert.class).getCertificate();
+                break;
+            }
+        } while (true);
```

Comment:
I'm always afraid to see `while (true)` loops in the code 😄 Could you refactor this to something a bit safer? Like `while (nbRetries < MAX_RETRIES)` or anything else?